src/main/java/recipeio/commands/FilterByAllergyCommand.java

```diff
@@ -20,17 +21,24 @@ public class FilterByAllergyCommand {
     public static void execute(String userInput, ArrayList<Recipe> recipes) {
 
         String allergy = InputParser.parseAllergyCriteria(userInput);
-
-        ArrayList<Recipe> matches = (ArrayList<Recipe>) recipes.stream()
-                .filter(recipe -> !recipe.allergies.contains(allergy))
-                .collect(Collectors.toList());
+        ArrayList<Integer> listNumbers = new ArrayList<>();
+        ArrayList<Recipe> matches = new ArrayList<>();
+
+        IntStream.range(0, recipes.size())
+                .forEach(i -> {
+                    Recipe recipe = recipes.get(i);
+                    if (!recipe.allergies.contains(allergy)) {
+                        matches.add(recipe);
+                        listNumbers.add(i + 1);
+                    }
+                });
```

Comment:
For findkeyowrd, you manage to implement this part using a simple for loop with if statement.

Let's do the same for here as well, so that it looks standardised.